File: DESIGN HASHMAP/Design_HashMap.java
import java.util.*;
class MyHashMap {
    private int size;
    ArrayList<ArrayList<int[]>> buckets;
    public MyHashMap() {
        this.size = 1009;
        this.buckets = new ArrayList<>();
        for(int i=0;i<size;i++){
            buckets.add(new ArrayList<>());
        }
    }
    public int _hash(int key){
        return key % size;
    }
    public void put(int key, int value) {
       int h = _hash(key);
       ArrayList<int[]>bucket = buckets.get(h);
       for(int[] pair : bucket){
        if(pair[0] == key){
            pair[1] = value;
            return;
        }
       }
       bucket.add(new int[]{key,value});
    }

    public int get(int key) {
       int h = _hash(key);
       ArrayList<int[]> bucket = buckets.get(h);
       for(int[] pair : bucket){
        if(pair[0] == key){
            return pair[1];
        }
       }
       return -1;
    }

    public void remove(int key) {
        int h = _hash(key);
        ArrayList<int[]> bucket = buckets.get(h);
        for(int i = 0;i<bucket.size();i++){
            if(bucket.get(i)[0] == key){
                bucket.remove(i);
                return;
            }
        }
    }
}

File: DESIGN HASHMAP/Design_HashMap.py
class MyHashMap:
    def __init__(self):
        # Your code here
        self.size = 1009
        self.buckets = [[] for _ in range(self.size)]
    def _hash(self,key):
        return key % self.size
    def put(self, key: int, value: int) -> None:
        # Your code here
        h = self._hash(key)
        bucket = self.buckets[h]
        for i,(k,v) in enumerate(bucket):
            if k == key:
                bucket[i] = (key,value)
                return
        bucket.append((key,value))

    def get(self, key: int) -> int:
        h = self._hash(key)
        bucket = self.buckets[h]
        for i ,(k,v) in enumerate(bucket):
            if k == key:
                return v
        return -1

    def remove(self, key: int) -> None:
        h = self._hash(key)
        bucket = self.buckets[h]
        for i, (k,v) in enumerate(bucket):
            if k == key:
                bucket.pop(i)
                return